### DISCUSSION TITLE
[预览] v2.3.1

### DISCUSSION THREAD

**Discussion by the1812** [the1812/Bilibili-Evolved#3461] [分享 / Share]
包含 v2.2.2 的所有更新内容.

✨新增
- `清爽首页` 的热门视频支持显示弹幕数量.
- `极简首页` 初版已完成.
- 脚本的更新源配置默认值更换为 `GitHub`, 并添加了 `AltCdn`, 表示开发者自定义的其他 CDN 源. `jsDelivr` 之后将会删除.
- 在 `GitHub` 更新源下, MDI 图标库更换使用 GitHub Pages.

☕开发者相关
- 重新整理了所有 CDN 调用, 并支持自定义 CDN 配置, 文档见 [doc/cdn.md](https://github.com/the1812/Bilibili-Evolved/blob/preview-features/doc/cdn.md).
- 拆分并整理了 `/src/core/settings` 的代码结构.
- 所有 webpack 配置更换为 TypeScript, 并启用 ESLint 检查.
- 重构了元数据注入功能, 新增了 i18n 和 description 注入, 详细用法可见 `webpack/inject-metadata/i18n.ts` 和 `webpack/inject-metadata/description.ts` 中的注释说明.
- 删除了用不到的包 `html-webpack-plugin`.
- 适配了 VSCode File Nesting 功能.

<hr /><em>This discussion was created from the release <a href='https://github.com/the1812/Bilibili-Evolved/releases/tag/v2.3.1-preview'>[预览] v2.3.1</a>.</em>

**Comment by bubbleboy11** [comment]
![image](https://user-images.githubusercontent.com/49860518/176099701-eb932e39-7774-43b1-98a0-5e5b0c0c9d6e.png)
关于的页面还是旧版

**Comment by GalaxySnail** [comment]
> 关于的页面还是旧版

可以复现：

- https://github.com/the1812/Bilibili-Evolved/blob/preview/dist/bilibili-evolved.preview.user.js#:~:text=v2.2.1%2Dpreview%2D1%2Dg86408879b%22%7D%7D%2C8168%3Afunction

在预览版脚本中搜索 `versionWithTag` 发现版本号仍然是 v2.2.1，不知道是构建的哪个步骤出了问题。

**Reply by the1812** [reply]
晚点我重新构建下，这确实有点怪，tag 明明已经打了的

**Reply by the1812** [reply]
重新发了个 v2.3.1

**Reply by GalaxySnail** [reply]
已解决🎉

不过现在的“关于”里面版本号不带 hash tag 了，对用户汇报 bug 来说会不会有些不方便

![](https://user-images.githubusercontent.com/53637421/176211849-e188a353-e96c-4434-b89a-3ede35c15628.png)

**Reply by the1812** [reply]
带的, `v2.3.1-preview` 本身对应一个 hash, 在这个 tag 之后还有别的 commit 才会带上后面的 hash
![image](https://user-images.githubusercontent.com/26504152/176212742-6f8c7662-d5ce-4b79-9119-7729df761716.png)

**Reply by GalaxySnail** [reply]
原来如此。明白了。

**Comment by hmgdd101** [comment]
大佬，新版的网址参数清理还是清不掉vdsource啊

**Reply by the1812** [reply]
应该也是版本号的问题, 再更新下, 现在最新的 commitHash 应该是 95fbb00c7 (预览版) / 095ddcc03 (正式版)

**Reply by hmgdd101** [reply]
目前还是清不掉，插件版本是Commit: dd5e26569，用的是v2.3.1-preview

**Reply by the1812** [reply]
更新源换成 GitHub 再去安装一次网址参数清理

**Comment by FoundTheWOUT** [comment]
简洁首页也太简洁了8

**Reply by the1812** [reply]
就是 v1 那味

**Reply by hmgdd101** [reply]
极简首页不能往下拉动态，加载不出后续来

**Reply by torch-jiu** [reply]
清爽和极简界面下,自定义顶栏会挡住刷新和上下页

**Comment by koyns** [comment]
清爽页面下为什么顶部横幅和顶栏都不见了怎么办？
